proof/proof.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package proof contains helpers for constructing log Merkle tree proofs.
+package proof
+
+import (
+	"errors"
+	"fmt"
+	"math/bits"
+
+	"github.com/transparency-dev/merkle/compact"
+)
+
+// Nodes contains information on how to construct a log Merkle tree proof.
```

Comment:
Probably worth expanding the comment here to say something along the lines that the proof built from these nodes can have at most 1 ephemeral node?